server/bleep/src/repo/iterator.rs

```diff
@@ -47,9 +47,18 @@ impl FileType {
     }
 }
 
+fn should_index_de(de: &ignore::DirEntry) -> bool {
+    should_index(&de.path())
+}
+
 fn should_index<P: AsRef<Path>>(p: &P) -> bool {
     let path = p.as_ref();
 
+    // TODO: Make this more robust
+    if path.components().any(|c| c.as_os_str() == ".git") {
+        return false;
+    }
```

Comment:
While this is true, it will break the test for `should_index` and introduce the constraint that any repo walking logic we use `should_index` with has to behave in the same way as `ignore::Walk`. So, although it's slightly inefficient would rather leave this check as is.